Extensions/video_downloader.js

```diff
@@ -0,0 +1,50 @@
+//* TITLE Video Downloader **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION	adds a download button to the video player **//
+//* DETAILS Adds a simple download button to all videos using the fancy player. Does not support 3rd-party players or the native player. Disabled on beta dash (now supported natively)**//
+//* DEVELOPER tlitookilakin **//
+//* FRAME false **//
+//* BETA false **//
+XKit.extensions.video_downloader = new Object({
+
+	running: false,
+
+	run: function() {
+		this.running = true;
+		if (!XKit.interface.is_tumblr_page() || XKit.page.react) {
+			return;
+		}
+		XKit.tools.init_css("video_downloader");
+		XKit.extensions.video_downloader.addButtons();
+		XKit.post_listener.add("video_downloader", XKit.extensions.video_downloader.addButtons);
+	},
+
+	destroy: function() {
+		XKit.post_listener.remove("video_downloader");
+		this.running = false;
+	},
+	
+	makeButton: function(url) {
+		var el = document.createElement("a");
+		var fname = url.split("/").pop();
+		el.classList.add("xvd-button");
+		el.setAttribute("href", url);
+		el.setAttribute("target", "_blank");
+		el.setAttribute("download", fname);
+		el.appendChild(document.createTextNode("Download this video"));
```

Comment:
`el.innerText = "Download this video"`?

Reply:
I can do that, but innerText is slower than just directly creating a text node
I can also do `el.appendChild(new Text("Download this video"));` if that's any clearer

Reply:
innerText is the clearest, even if it's technically less performant